### PR TITLE
Enable SSL for http server

### PR DESCRIPTION
**Pull request by sachaaaaa**
The ssl stuff is mostly from boost beast async_ssl example.
The certificate generation is mostly from mkcert.c (https://opensource.apple.com/source/OpenSSL/OpenSSL-22/openssl/demos/x509/mkcert.c)

`server_certificate.h` is still pretty rough but as a temporary thing it should be ok.

Note: this PR will break snode to snode requests (httpClientSession needs to use ssl as well). Another PR is coming up for that.

### CODE REVIEW

**Comment by sachaaaaa** [comment]
The client side needs to be redone as I just realised we are doing http requests to lokid that should not use the ssl context....

**Comment by sachaaaaa** [comment]
Re-requesting reviews since I've added the client side ssl (for snode to snode only).

Integration tests have passed but requires https://github.com/msgmaxim/swarm_migration_testing/pull/1

File: httpserver/CMakeLists.txt [hunk 1]
```diff
@@ -10,7 +10,8 @@ set(HEADER_FILES
     service_node.h
     serialization.h
     rate_limiter.h
-    ../external/json.hpp)
+    ../external/json.hpp
+    https_client.h)
```
Comment (msgmaxim) [inline httpserver/CMakeLists.txt]: Should also add `server_certificates.h`?

File: httpserver/server_certificates.h [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    file.close();
+
+    file.open(dh_path);
+    const std::string dh((std::istreambuf_iterator<char>(file)),
+                         std::istreambuf_iterator<char>());
+    file.close();
+
+    ctx.set_password_callback(
+        [](std::size_t, boost::asio::ssl::context_base::password_purpose) {
+            return "test";
```
Comment (msgmaxim) [inline httpserver/server_certificates.h]: Is this OK?

File: httpserver/server_certificates.h [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+#include <openssl/evp.h>
+#include <openssl/pem.h>
+#include <openssl/x509v3.h>
+
+#include <cstddef>
+#include <fstream>
+#include <iostream>
+#include <memory>
+
+int callback(int p, int n, void* arg) {
```
Comment (msgmaxim) [inline httpserver/server_certificates.h]: Should be removed.

File: httpserver/server_certificates.h [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+void generate_dh_pem(const char* dh_path) {
+    const int prime_len = 2048;
+    const int generator = DH_GENERATOR_2;
+    DH* dh = DH_new();
+    if (dh == NULL) {
+        printf("Alloc for dh failed\n");
+        ERR_print_errors_fp(stderr);
+        abort();
+    }
+    std::cout << "Generating DH parameter, this might take a while..."
```
Comment (msgmaxim) [inline httpserver/server_certificates.h]: Use BOOST_LOG?

Comment (sachaaaaa) [reply httpserver/server_certificates.h]: Done

File: httpserver/server_certificates.h (hunk 3 above)
Comment (sachaaaaa) [reply httpserver/server_certificates.h]: Removed

File: httpserver/server_certificates.h (hunk 2 above)
Comment (sachaaaaa) [reply httpserver/server_certificates.h]: Not used (the keys are not protected) so removed

File: httpserver/CMakeLists.txt (hunk 1 above)
Comment (sachaaaaa) [reply httpserver/CMakeLists.txt]: Done, for consistency.

**Comment by sachaaaaa** [comment]
Closing since github went bananas

**Pull request closed without merging**